text/0052-tx-summary-digest.md

```diff
@@ -0,0 +1,510 @@
+* Now having the amount in hand, the next step is to check the `has_input_rules` flag from `TxInSummary`. We know that
+  this flag is accurate because the consensus enclave checks the `TxSummary` and can see which inputs actually have rules.
+  If there are no rules, then we know that we must be signing for this transaction. If there are rules then we attribute
+  the input to an anonymous swap counterparty.
+
+#### Assumptions around `InputRules`
+
+There is an assumption which is worth explaining here, that inputs and outputs that match to `InputRules` never belong to the person signing the transaction.
+
+This is reasonable because it is never sensible for a person to be a counterparty to their own Signed Contingent Input. Conceptually, an SCI is a signed input
+with "strings attached", and the `InputRules` are those strings. So the nature of the interaction is "here is my input which you can use, as long as you
+follow these rules". `InputRules` can only be used
+to constrain the person who uses a particular signed input in a transaction. If you are in fact the signer of the input, you could always remove the input
+rules and sign the input normally, and your transaction would be valid if it were valid before (in strictly more cases). Offering yourself one of your own
+inputs "with strings attached" never makes sense, because you already had the input and could just sign it again without the strings.
+
+So, a real user would generally prefer not to make their own input into an SCI and then spend it in a transaction, for simplicity.
+This observation applies no matter what future kinds of `InputRules` we create.
+
+However, because of the threat model, we may be concerned that a malicious host computer would arbitrarily attach empty input rules to the user's inputs,
+in order to confuse the hardware wallet signing device, and cause it to display "self" inputs as "swap counterparty" inputs. And could similarly do so for
+outputs.
+
+The following analysis explains why these attempts fail:
+* For `TxOut`'s, the decision tree is specified so that we view-key match it first, and if it matches to use, we short circuit and declare it to be
+  a credit to ourself, regardless of whether it is associated to input rules.
+* For inputs, simply attaching input rules to an input, and then later asking the device to sign it as if there are no input rules, will not work.
+  In MCIP 31, inputs that have associated input rules have to sign a different digest that does not depend on the entire Tx. If instead the entire Tx
+  digest is signed, then the transaction will fail validation.
+
+There is an additional possibility: Suppose that the hardware wallet has features that also support signing the MCIP 31 SCIs. Then it could be possible
+that we authorize an sign an SCI, and then later we build a transaction that matches against it, so that we are essentially "trading with ourselves".
```

Comment:
```suggestion
that we authorize and sign an SCI, and then later we build a transaction that matches against it, so that we are essentially "trading with ourselves".
```